module/VuFind/src/VuFind/View/Helper/Root/MakeLink.php

```diff
@@ -0,0 +1,109 @@
+<?php
+/**
+ * Make link view helper
+ *
+ * PHP version 7
+ *
+ * Copyright (C) Villanova University 2020.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+namespace VuFind\View\Helper\Root;
+
+use Laminas\View\Helper\AbstractHelper;
+
+/**
+ * Make link view helper
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class MakeLink extends AbstractHelper
+{
+    /**
+     * Create an anchor tag
+     *
+     * If $href, will try to find an href in $attrs
+     * > makeLink('text', ['href' => '#', 'class' => 'btn-link'])
+     *
+     * If $attrs is a string, will be treated like a class
+     * > makeLink('text', 'btn-link') == makeLink('text', ['class' => 'btn-link'])
+     *
+     * @param string       $text  Link contents
+     * @param string|array $href  Link destination (skippable for (text, attrs))
+     * @param array        $attrs Link attributes (associative array)
+     *
+     * @return string HTML for an anchor tag
+     */
+    public function __invoke($text, $href = null, $attrs = [])
+    {
+        // Class string
+        if (is_string($attrs)) {
+            $attrs = ['class' => $attrs];
+        }
+
+        // Skipped $href, treat $href like $attrs
+        if (is_array($href)) {
+            $attrs = $href;
+            $href = null;
+        }
+
+        // Plain text on falsey href
+        if (!$href && !isset($attrs['href'])) {
+            // Don't waste passed in attributes
+            if (!empty($attrs)) {
+                return $this->compileAttrs($text, $attrs, 'span');
+            }
+
+            return $text;
+        }
+
+        // Compile attributes
+        if ($href && !isset($attrs['href'])) {
```

Comment:
I wonder if this would be less confusing if the code were refactored/rearranged a bit -- process all the inputs at the top, and then perhaps there could be fewer return statements, and they could be grouped together closer to the bottom. Happy to take a stab at it if you want to see what I have in mind.

Reply:
Also, what is the expected behavior when both $href and $attrs['href'] are defined? Right now it seems that $attrs['href'] takes precedence... but do we need a warning or exception? A test case?